### DISCUSSION TITLE
Replacement Whitebox T3 arrived... not the results I was hoping for

### DISCUSSION THREAD

**Discussion by johnny2678** [tagyoureit/nodejs-poolController#996] [Q&A] [answered]
I decided to stop clobbering @flamike's thread and start my own.

My replacement T3 arrived today.  I guess this is a new board revision because It says mkii on the back and sports sma connectors instead of BNC.

![Untitled](https://github.com/tagyoureit/nodejs-poolController/assets/8219743/e00fb143-c38e-486a-9d3f-a27979b87c89)
![Untitled 2](https://github.com/tagyoureit/nodejs-poolController/assets/8219743/2a4e1327-5630-418a-8cf6-7e618041e03a)

To catch everyone up, my probes readings were intermittently going out.  I'd get sporadic readings that would go silent for hours, sometimes days at a time:
![image](https://github.com/tagyoureit/nodejs-poolController/assets/8219743/78f24bec-9cf0-4cca-91a0-151a13879c6b)

At first I thought of the flow sensor, because I have it set to only take readings when there's flow... but that was working.

When I checked the REM logs, it confirmed that no readings were seen, but nothing to indicate why:
```
1|REM  | [09/07/2024, 17:08:33] warn: Atlas_EZO-ORP could not retrieve status result was empty
1|REM  | [09/07/2024, 17:08:33] warn: Atlas_EZO-pH could not retrieve status result was empty
1|REM  | [09/07/2024, 17:08:34] error: Atlas_EZO-ORP Discarded invalid value from I2C result:undefined
1|REM  | [09/07/2024, 17:08:34] error: Atlas_EZO-pH Discarded invalid value from I2C result:undefined
1|REM  | [09/07/2024, 17:08:40] warn: Atlas_EZO-ORP could not retrieve status result was empty
1|REM  | [09/07/2024, 17:08:40] warn: Atlas_EZO-pH could not retrieve status result was empty
1|REM  | [09/07/2024, 17:08:41] info: [5:08:41 PM] 192.168.5.148 GET /state/device/i2c:1:3:in0_10.8 {}
1|REM  | [09/07/2024, 17:08:41] info: [5:08:41 PM] 192.168.5.148 GET /status/device/i2c:1:1 {}
1|REM  | [09/07/2024, 17:08:41] info: [5:08:41 PM] 192.168.5.148 GET /status/device/i2c:1:2 {}
1|REM  | [09/07/2024, 17:08:41] info: [5:08:41 PM] 192.168.5.148 GET /status/device/i2c:1:6:1 {}
1|REM  | [09/07/2024, 17:08:41] info: [5:08:41 PM] 192.168.5.148 PUT /state/device/i2c:1:6:1 {"state":false}
1|REM  | [09/07/2024, 17:08:41] info: Setting device state {"type":"i2c","busId":1,"deviceId":6,"binding":"i2c:1:6:1","typeId":1,"id":6,"params":["1"]}
1|REM  | setDeviceState
1|REM  | [09/07/2024, 17:08:41] info: [5:08:41 PM] 192.168.5.148 PUT /state/device/i2c:1:6:1 {"state":false}
1|REM  | [09/07/2024, 17:08:41] info: Setting device state {"type":"i2c","busId":1,"deviceId":6,"binding":"i2c:1:6:1","typeId":1,"id":6,"params":["1"]}
1|REM  | setDeviceState

```

Then I thought to check i2cdetect, sure enough, 62 (orp) and 63 (pH) were missing from the output:
```
pi@PiPoolv2:~ $ sudo i2cdetect -y 1
     0  1  2  3  4  5  6  7  8  9  a  b  c  d  e  f
00:          03 -- -- -- -- -- -- -- -- -- -- -- -- 
10: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- 
20: -- -- -- -- -- -- -- 27 -- -- -- -- -- -- -- -- 
30: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- 
40: -- -- -- -- -- -- -- -- 48 -- -- -- -- -- -- -- 
50: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- 
60: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- 
70: -- -- -- -- -- -- -- --    
```
03 = Sequent SmartFan
27 = Multi Relay Board
48 = MEGA-BAS

This is what led me to think that maybe my T3 had fried.  AFter all, it has been sitting in the west facing FL sun outside for 2 years.

Atlas had sold out of T3 boards but luckily, [whitebox still sells them direct](https://www.whiteboxes.ch/shop/whitebox-t3-for-raspberry-pi/), and at around 66% or the Atlas price! For $36 I figure, "why not?" and order.  10 short days later the package from Switzerland showed up.

I pulled the old T3 and swapped the new one in, fired back up the pi stack, AND!... nothing.  Same deal... probe chips weren't detected by i2cdetect and I don't have ph/orp readings.  The chip lights do not come on.

So I'm stuck.  There isn't really any t3 board configuration to play with.  It's just plug and play. I have a hard time believing that both my sensor chips fried at the same time, but I guess I could try replacing those as well?  Is it possible that some pin used by the whitebox is fried on the pi or somewhere else down the stack?

I dunno, just looking for any ideas.

**Comment by FlaMike** [comment]
Some straws I'd grasp for starters...

Can you check voltages on the Pi stack pins at the T3 board and one board below to make sure you have power available?  If that's OK, I guess I'd check the pins on the EZO boards as well.  No lights, no phones, no motor cars, it's as frightening as can be?

BTW, I'm flattered that you saw fit to clobber my thread :-)  I mean that is the most grateful way.

Good luck & there's a pin along the stack that's messed up but repairable???

I'm getting ready to install at 24 VDC transformer that *could* show up tomorrow so I can get my pool back too.  I hope you beat me to the punch of pool bliss!

**Comment by rstrouse** [comment]
So there are no lights on the chips at all even during boot.  I would install only the T3 on the pi then check to see if i2cdetect finds the chips.

**Reply by johnny2678** [reply]
> So there are no lights on the chips at all even during boot

Correct.

I might try testing it by hooking it up to a new pi so I don't have to unscrew my entire stack.

**Reply by FlaMike** [reply]
Good plan.  Pad ballet, especially when taking a Pi stack apart, is never fun.  Good luck!!!

**Reply by tagyoureit** [reply]
Is the T3 even getting power?  I would try to turn on your stack minus the ph/orp sensors and put a voltage meter across the vcc/gnd on the T3 to see if there is any juice there.

Or try with the chips in and see if you get any readings.  (Oh, nvm... I think the soldering is under the protective covering.)

**Reply by johnny2678** [reply]
> Is the T3 even getting power? I would try to turn on your stack minus the ph/orp sensors and put a voltage meter across the vcc/gnd on the T3 to see if there is any juice there.

You mean try and probe the vcc/gnd leads to the chip with the chip removed?  I think I could do that.  

Otherwise, I'm not sure which pins give the t3 power/gnd

**Comment by johnny2678** [comment, marked as answer]
I don't want to jinx it but... I have ph/orp readings!!

![image](https://github.com/tagyoureit/nodejs-poolController/assets/8219743/0d41a21b-bcce-4cac-a018-ed9c799f5f98)

I doubt you pros would fall into this trap but still dropping this here in case it helps anyone else.

So I took a spare rPi3 I had lying around out to my pad and plugged the new T3 in.  It was dead.  No probe chip LEDs or LEDs on the T3 board.  

At this point, I was really confused.  It's brand new.  Then, as I was wiggling it back off the GPIO header pins on the spare pi, the LEDs flashed.

Aha... I had to get out the magnifying glass.  The new T3 board came with some type of spring pins pre-installed that you can push male GPIO pins through.  On the top of the board, there is a 2-3 mm riser above the PCB layer.  

I didn't take a pic of how it looked when it came. I should have.  But these are what the spring pins look like under the riser.
![image](https://github.com/tagyoureit/nodejs-poolController/assets/8219743/ebf2e8be-ed45-4b10-a91d-c71a20fdc3fb)

I had never seen this before.  I'm guessing the riser is to hold the spring pins in place?  But I didn't know that at the time and I needed the GPIO pin height to make sure I had good contact with the next board upt he stack.  

You can see the problem.  The top 2 pins and 1 bottom pin were ripped off when I did this.  

Found them!
![image](https://github.com/tagyoureit/nodejs-poolController/assets/8219743/fa6201cb-a3d2-40c4-80ba-f7e90b91323c)

Obviously the pi stack pins that were missing provide power and weren't able to make contact:
![image](https://github.com/tagyoureit/nodejs-poolController/assets/8219743/b940a72c-975f-47ca-8a3f-4be92fb247b8)


A quick solder job later and I am back in business!

side note: the old t3 didn't work on the spare pi either, and I did check all the contacts.. I think it really was fried. So glad I ordered a spare.

**Reply by FlaMike** [reply]
Very strange, but happy you were able to identify the problem, fix it, & are now back in business!!!

If you're bored sometime, you can check out my seemingly never-ending challenge.  I am *so* confused by mine! 

I hope you can enjoy your success for years & years!
